segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/StreamSegmentContainer.java

```diff
@@ -714,4 +728,74 @@ private void shutdownWhenStopped(Service component, String componentName) {
     }
 
     //endregion
+
+    //region DirectSegmentWrapper
+
+    /**
+     * Direct Segment Access implementation.
+     */
+    @RequiredArgsConstructor
+    private class DirectSegmentWrapper implements DirectSegmentAccess {
+        @Getter
+        private final long segmentId;
+
+        @Override
+        public CompletableFuture<Long> append(byte[] data, Collection<AttributeUpdate> attributeUpdates, Duration timeout) {
+            ensureRunning();
+            StreamSegmentAppendOperation operation = new StreamSegmentAppendOperation(this.segmentId, data, attributeUpdates);
```

Comment:
Compared to the workflow of `StreamSegmentContainer`, these operations are neither logging the request nor updating the metrics. Regarding the former point, would it be useful for debugging purposes to identify requests being served via a `DirectSegmentWrapper`? Regarding the latter point, I feel that if we perform many operations via `DirectSegmentWrapper` we may loose significant information on the actual load of the container.

Reply:
1. Good point on logging. I added the `logRequest` for each of those calls (that deserved it). We can tell the difference between the two paths since the original logs segment name while `DirectSegmentAccess` logs segment ids (since that's what's available).
2. Metrics are not recorded in `SegmentContainer`. They are one level up (`StreamSegmentService`), and are designed to track external requests. `DirectSegmentAccess` is meant to be an internal request, so no need to pollute metrics with these.